### DISCUSSION TITLE
How to add a custom method on a mongoose Model

### DISCUSSION THREAD

**Discussion by samislam** [Automattic/mongoose#12289] [Q&A] [answered]
In Mongoose, we can add methods to documents as follows:
```js
const userSchema = new mongoose.Schema({
  balance: Number
})
userSchema.methods.withdrawBalance = function(amount){
  const doc = this
  doc.balance = doc.balance - amount
}
```
and we can use it as follows:
```js
const userDoc = UserModel.findById(ID_HERE);
userDoc.deductBalance(100); // -100$ USD
userDoc.save()
```
I want to be able to do the same on the models the same way I can do it on documents, I want to be able to do something like that:
```js
const userDoc = UserModel.findById(ID_HERE).myCustomMethod()
```
But How?

**Comment by vkarpov15** [comment, marked as answer]
Query helpers. See:

1. Mongoose docs: https://mongoosejs.com/docs/guide.html#query-helpers
2. My blog: http://thecodebarbarian.com/mongoose-custom-query-methods.html
